internal/noderesourcetopology/equality_test.go

```diff
@@ -0,0 +1,132 @@
+ * Copyright 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package noderesourcetopology
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	nrtv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+)
+
+func TestEqualZones(t *testing.T) {
+	testCases := []struct {
+		name     string
+		data1    nrtv1alpha2.ZoneList
+		data2    nrtv1alpha2.ZoneList
+		expected bool
+	}{
+		{
+			name: "sorted",
+			data1: nrtv1alpha2.ZoneList{
+				{
+					Name: "Zone-000",
+					Resources: nrtv1alpha2.ResourceInfoList{
+						{
+							Name:        "foo",
+							Capacity:    resource.MustParse("2"),
+							Allocatable: resource.MustParse("2"),
+							Available:   resource.MustParse("2"),
+						},
+						{
+							Name:        "bar",
+							Capacity:    resource.MustParse("4"),
+							Allocatable: resource.MustParse("4"),
+							Available:   resource.MustParse("4"),
+						},
+					},
+				},
+				{
+					Name: "Zone-001",
+					Resources: nrtv1alpha2.ResourceInfoList{
+						{
+							Name:        "bar",
+							Capacity:    resource.MustParse("4"),
+							Allocatable: resource.MustParse("4"),
+							Available:   resource.MustParse("4"),
+						},
+						{
+							Name:        "foo",
+							Capacity:    resource.MustParse("2"),
+							Allocatable: resource.MustParse("2"),
+							Available:   resource.MustParse("2"),
+						},
+					},
+				},
+			},
+			data2: nrtv1alpha2.ZoneList{
+				{
+					Name: "Zone-001",
+					Resources: nrtv1alpha2.ResourceInfoList{
+						{
+							Name:        "foo",
+							Capacity:    resource.MustParse("2"),
+							Allocatable: resource.MustParse("2"),
+							Available:   resource.MustParse("2"),
+						},
+						{
+							Name:        "bar",
+							Capacity:    resource.MustParse("4"),
+							Allocatable: resource.MustParse("4"),
+							Available:   resource.MustParse("4"),
+						},
+					},
+				},
+				{
+					Name: "Zone-000",
+					Resources: nrtv1alpha2.ResourceInfoList{
+						{
+							Name:        "foo",
+							Capacity:    resource.MustParse("2"),
+							Allocatable: resource.MustParse("2"),
+							Available:   resource.MustParse("2"),
+						},
+						{
+							Name:        "bar",
+							Capacity:    resource.MustParse("4"),
+							Allocatable: resource.MustParse("4"),
+							Available:   resource.MustParse("4"),
+						},
+					},
+				},
+			},
+			expected: true,
```

Comment:
I think it's nearly enough to test this case. 
just to clarify, don't we want usually to add negative case chack, like the expected to be false, just to verify the equality check also works for the inner items as expected?

Reply:
good point. Yes we should add at least a negative case